src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -27557,7 +27557,7 @@ void M(string? s)
         [Fact]
         public void NotNullIfNotNull_Return()
         {
-            // NotNullIfNotNull isn't enforced within method bodies
+            // NotNullIfNotNull isn't enforced in scenarios like the following
```

Comment:
>// NotNullIfNotNull isn't enforced in scenarios like the following [](start = 12, length = 66)

nit: the comment probably should be removed. Since parameter `p` is maybe-null upon exit, the problem isn't that we didn't enforce NotNullIfNotNull.

Reply:
I guess the comment is saying: if our analysis were more rigorous, we would warn here. But we can't distinguish scenarios like the following, so we don't warn in either place:

```cs
[return: NotNullIfNotNull("x")]
X? M1(X? x) => null; // ideally would warn here--what if 'x' actually contains not-null?

[return: NotNullIfNotNull("x")]
X? M2(X? x)
{
    if (x is null)
    {
        return null; // shouldn't warn here--'x' is always null.
    }

    return x;
}
```

Reply:
Thanks for the clarification. Makes sense